src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreDatabaseTaskDataObject.cs

```diff
@@ -664,6 +664,23 @@ private void UpdateDbFiles()
         }
 
         /// <summary>
+        /// Combining the root and file name using the server connection file path seperator
+        /// </summary>
+        private string CombineToServerConnectionPath(string root, string filePath) 
+        {
+            string pathSeparatorFromServerConnection = PathWrapper.PathSeparatorFromServerConnection(Server.ConnectionContext);
+            string combinedPath = PathWrapper.Combine(root, filePath);
+            
+            // Make sure all path seperators are server connection separator
+            string result = combinedPath.Replace(Path.DirectorySeparatorChar.ToString(), pathSeparatorFromServerConnection);
+
+            // Make sure there's not any double file seperator in the path
+            result = result.Replace(pathSeparatorFromServerConnection + pathSeparatorFromServerConnection, 
```

Comment:
nit: line 679 should probably have another tab to make it clearer that is parameter to .Replace